tests/segmentation/test_threshold.py

```
@@ -185,3 +185,32 @@ def test_thresholds_too_too_few_thresholds(self):
+
+    def test_with_custom_column_names(self):
+        """Test ThresholdSegmentation with custom column names."""
+        df = pd.DataFrame(
+            {
+                "buyer_id": [1, 2, 3, 4, 5, 1, 2, 3],
+                "spend_amount": [100, 200, 50, 150, 75, 50, 100, 300],
+                "purchase_date": pd.date_range("2023-01-01", periods=8),
+            },
+        )
+
+        custom_columns = {
```

Ruff (0.11.9)

Local variable `custom_columns` is assigned to but never used

Remove assignment to unused variable `custom_columns`

(F841)